src/ui/webeiddialog.cpp

```diff
@@ -81,29 +80,53 @@ WebEidDialog::WebEidDialog(QWidget* parent) : WebEidUI(parent), ui(new Private)
     }
     setWindowFlag(Qt::CustomizeWindowHint);
     setWindowFlag(Qt::WindowTitleHint);
-    setWindowTitle(qApp->applicationDisplayName());
-    setTrText(ui->aboutVersion, [] { return tr("Version: %1").arg(qApp->applicationVersion()); });
-    QMenu* langMenu = new QMenu(ui->langButton);
-    langMenu->addAction(QStringLiteral("EN"));
-    langMenu->addAction(QStringLiteral("ET"));
-    langMenu->addAction(QStringLiteral("FI"));
-    langMenu->addAction(QStringLiteral("HR"));
-    langMenu->addAction(QStringLiteral("RU"));
-    connect(langMenu, &QMenu::triggered, qApp, [this](QAction* action) {
-        QSettings().setValue(QStringLiteral("lang"), action->text().toLower());
-        ui->langButton->setText(action->text());
-        qApp->loadTranslations();
-    });
-#ifdef Q_OS_LINUX
-    setStyleSheet(styleSheet() + QStringLiteral("#langButton {padding-right: 15px;}"));
-    connect(ui->langButton, &QToolButton::clicked, this, [this, langMenu] {
-        langMenu->exec(mapToGlobal(ui->langButton->geometry().bottomLeft()));
+    setWindowTitle(QApplication::applicationDisplayName());
+    setTrText(ui->aboutVersion,
+              [] { return tr("Version: %1").arg(QApplication::applicationVersion()); });
+
+    ui->langButton = new QToolButton(this);
+    ui->langButton->setObjectName("langButton");
+    static const QMap<QString, QString> LANG_LIST {
+        {QStringLiteral("et"), QStringLiteral("Eesti")},
+        {QStringLiteral("en"), QStringLiteral("English")},
+        {QStringLiteral("ru"), QStringLiteral("Русский")},
+        {QStringLiteral("fi"), QStringLiteral("Finnish")},
+        {QStringLiteral("hr"), QStringLiteral("Hrvatska")}};
+    ui->langButton->setText(tr("EN"));
+    ui->langButton->setAccessibleName(LANG_LIST.value(ui->langButton->text().toLower()));
+    connect(ui->langButton, &QToolButton::clicked, this, [this] {
+        if (auto* menu = findChild<QWidget*>(QStringLiteral("langMenu"))) {
+            menu->deleteLater();
+            return;
+        }
+        auto* menu = new QWidget(this);
+        menu->setObjectName("langMenu");
+        auto* layout = new QVBoxLayout(menu);
+        layout->setContentsMargins(1, 1, 1, 1);
+        layout->setSpacing(1);
+        auto* langGroup = new QButtonGroup(menu);
+        langGroup->setExclusive(true);
+        for (auto i = LANG_LIST.cbegin(); i != LANG_LIST.cend(); ++i) {
+            auto* action = new QPushButton(menu);
+            action->setText(i.value());
+            action->setProperty("lang", i.key());
+            action->setAutoDefault(false);
+            layout->addWidget(action);
+            langGroup->addButton(action);
+            action->setCheckable(true);
+            action->setChecked(i.key() == ui->langButton->text().toLower());
```

Comment:
If you want to get fancy, you could use a range-based for loop with structured binding here:

```suggestion
        for (auto&& [key, value] : asKeyValueRange(LANG_LIST)) {
            auto* action = new QPushButton(menu);
            action->setText(value);
            action->setProperty("lang", key);
            action->setAutoDefault(false);
            layout->addWidget(action);
            langGroup->addButton(action);
            action->setCheckable(true);
            action->setChecked(key == ui->langButton->text().toLower());
```

However, we need to bring in `asKeyValueRange()` from https://stackoverflow.com/a/64985764/258772 in this case:
```c++
template <typename T>
class asKeyValueRange
{
public:
    asKeyValueRange(T& data) : m_data{data} {}

    auto begin() { return m_data.keyValueBegin(); }

    auto end() { return m_data.keyValueEnd(); }

private:
    T& m_data;
};
```

Reply:
I hope Qt will resolve this one day.